rotkehlchen/chain/bitcoin/manager.py

```diff
@@ -522,6 +522,48 @@ def _decode_single_side_transfers(
 
         return spend_events + receive_events  # keep all spend events first
 
+    def _decode_many_to_many_transfers(
+            self,
+            tx: BitcoinTx,
+            inputs: dict[BTCAddress, FVal],
+            outputs: dict[BTCAddress, FVal],
+    ) -> list[HistoryEvent]:
+        """Decode transfers in a multi-input, multi-output transaction.
+        There isn't a direct A sent to B. Instead it's all A's sent to all B's. So we can only
+        decode the total amount each input address sent to all the outputs and the total amount
+        each output address received from all the inputs.
```

Comment:
i think this docstring needs tweaking; the method name is less confusing which shouldn't be the case.